src/query/storages/fuse/src/operations/read/runtime_filter_prunner.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use common_catalog::plan::PartInfoPtr;
+use common_exception::Result;
+use common_expression::ConstantFolder;
+use common_expression::Expr;
+use common_expression::FunctionContext;
+use common_expression::Scalar;
+use common_expression::TableSchema;
+use common_functions::BUILTIN_FUNCTIONS;
+use log::info;
+use storages_common_index::statistics_to_domain;
+
+use crate::FusePartInfo;
+
+pub fn runtime_filter_pruner(
```

Comment:
This should not be in part's level, because the part is lazy and its `columns_stat` is not accurate enough.

I suggest to apply this in `FusePruner` ( `SegmentPruner` and `block_pruner`)

Reply:
I mis understand.